scripts/cli/cmd/deployment.js

```diff
@@ -30,7 +30,8 @@ async function deployRelayerContract(cfg) {
         console.log("[Relayer] Transaction Hash: ", contract.deployTransaction.hash);
         await contract.deployed();
     } catch (e) {
-        throw e
+        console.log(e)
```

Comment:
Why this change? I tested it and `throw e` was exiting the tests and causing it to fail (with a reason), was it not?

Reply:
Wasnt't sure of the behaviour tbh. Isn't throwing it the same as not having a try/catch?

Reply:
Well the try part will catch any failures to deploy, which you found previously that it was failing silently, so the `throw e` will escalate the error and cause the tests to exit with I think an exit code of 1 (either way the tests fail when the throw is triggered)

Reply:
it was when I ran the scripts themselves that I had issues previously, sounds like you were running the tests? Perhaps the exit code is returned by truffle and not the script itself?

Reply:
So I ran `./scripts/cli/index.js deploy --port=8545` with the local ganache running, the `chainID` argument missing from `deployBridgeContract`, and just a `console.log(e)` in the catch block, and I get the silent error you mentioned:
```bash
[Relayer] Initial Relayers:  [ '0xff93B45308FD417dF303D6515aB04D9e89a750Ca',
  '0x8e0a907331554AF72563Bd8D43051C2E64Be5d35',
  '0x24962717f8fA5BA3b931bACaF9ac03924EB475a0',
  '0x148FfB2074A9e59eD58142822b3eB3fcBffb0cd7',
  '0x4CEEf6139f00F9F4535Ad19640Ff7A0137708485' ]
[Relayer] Relayer Threshold:  2
[Relayer] Contract address:  0xd7E33e1bbf65dC001A0Eb1552613106CD7e40C31
[Relayer] Transaction Hash:  0xe35ff550cb120cf28da19b579c1d7fb0369273053387563dcec669052bed1a2c
{ Error: missing argument in Contract constructor (count=2, expectedCount=3, version=4.0.46)
    at throwError (/home/anon/code/ChainSafe/git-repos/chainbridge-solidity/node_modules/ethers/errors.js:76:17)
    at Object.checkArgumentCount (/home/anon/code/ChainSafe/git-repos/chainbridge-solidity/node_modules/ethers/errors.js:96:9)
    at ContractFactory.getDeployTransaction (/home/anon/code/ChainSafe/git-repos/chainbridge-solidity/node_modules/ethers/contract.js:669:16)
    at ContractFactory.deploy (/home/anon/code/ChainSafe/git-repos/chainbridge-solidity/node_modules/ethers/contract.js:681:44)
    at Object.deployBridgeContract (/home/anon/code/ChainSafe/git-repos/chainbridge-solidity/scripts/cli/cmd/deployment.js:43:38)
    at Command.<anonymous> (/home/anon/code/ChainSafe/git-repos/chainbridge-solidity/scripts/cli/index.js:35:22)
    at process._tickCallback (internal/process/next_tick.js:68:7)
  reason: 'missing argument in Contract constructor',
  code: 'MISSING_ARGUMENT',
  count: 2,
  expectedCount: 3 }
[ERC20 Handler] Contract address:  0xc279648CE5cAa25B9bA753dAb0Dfef44A069BaF4
[ERC20 Handler] Transaction Hash:  0xc58e4f34580effe82bfd36309f195633a5f1b479800752c1be733585a40d04f8
[ERC20 Token] Contract address:  0x84b141Aada70e2B0C3Ec25d24E81032328ea1b1A
[ERC20 Token] Transaction Hash:  0xeddbf7790c9e29e0617ee4df8b2508b5d8eb5fe6fac41dd53c88f4c5b32fccb9
```

Running the same tests with `chainID` missing but replacing the `console.log(e)` with `throw e` I get
```bash
[Relayer] Initial Relayers:  [ '0xff93B45308FD417dF303D6515aB04D9e89a750Ca',
  '0x8e0a907331554AF72563Bd8D43051C2E64Be5d35',
  '0x24962717f8fA5BA3b931bACaF9ac03924EB475a0',
  '0x148FfB2074A9e59eD58142822b3eB3fcBffb0cd7',
  '0x4CEEf6139f00F9F4535Ad19640Ff7A0137708485' ]
[Relayer] Relayer Threshold:  2
[Relayer] Contract address:  0x771ce6cda91d18eB76029d17aCcd85834F5C0303
[Relayer] Transaction Hash:  0xc76234ffb12c476270bb732f8bf4e6107d58dad281e618ce5fac18fc796b24b9
(node:9097) UnhandledPromiseRejectionWarning: Error: missing argument in Contract constructor (count=2, expectedCount=3, version=4.0.46)
    at throwError (/home/anon/code/ChainSafe/git-repos/chainbridge-solidity/node_modules/ethers/errors.js:76:17)
    at Object.checkArgumentCount (/home/anon/code/ChainSafe/git-repos/chainbridge-solidity/node_modules/ethers/errors.js:96:9)
    at ContractFactory.getDeployTransaction (/home/anon/code/ChainSafe/git-repos/chainbridge-solidity/node_modules/ethers/contract.js:669:16)
    at ContractFactory.deploy (/home/anon/code/ChainSafe/git-repos/chainbridge-solidity/node_modules/ethers/contract.js:681:44)
    at Object.deployBridgeContract (/home/anon/code/ChainSafe/git-repos/chainbridge-solidity/scripts/cli/cmd/deployment.js:43:38)
    at Command.<anonymous> (/home/anon/code/ChainSafe/git-repos/chainbridge-solidity/scripts/cli/index.js:35:22)
    at process._tickCallback (internal/process/next_tick.js:68:7)
(node:9097) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 2)
(node:9097) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```
And for comparison, this is what I get when running the same with your changes

```bash
  '0x8e0a907331554AF72563Bd8D43051C2E64Be5d35',
  '0x24962717f8fA5BA3b931bACaF9ac03924EB475a0',
  '0x148FfB2074A9e59eD58142822b3eB3fcBffb0cd7',
  '0x4CEEf6139f00F9F4535Ad19640Ff7A0137708485' ]
[Relayer] Relayer Threshold:  2
[Relayer] Contract address:  0x39863e3eDB5255dB93bBf8E76c12578357dBe6c7
[Relayer] Transaction Hash:  0x6419cc0dd833b00767f480049bb36179d7cc2d7dd316d6a2842fd2b90e40e8a6
{ Error: missing argument in Contract constructor (count=2, expectedCount=3, version=4.0.46)
    at throwError (/home/anon/code/ChainSafe/git-repos/chainbridge-solidity/node_modules/ethers/errors.js:76:17)
    at Object.checkArgumentCount (/home/anon/code/ChainSafe/git-repos/chainbridge-solidity/node_modules/ethers/errors.js:96:9)
    at ContractFactory.getDeployTransaction (/home/anon/code/ChainSafe/git-repos/chainbridge-solidity/node_modules/ethers/contract.js:669:16)
    at ContractFactory.deploy (/home/anon/code/ChainSafe/git-repos/chainbridge-solidity/node_modules/ethers/contract.js:681:44)
    at Object.deployBridgeContract (/home/anon/code/ChainSafe/git-repos/chainbridge-solidity/scripts/cli/cmd/deployment.js:43:38)
    at Command.<anonymous> (/home/anon/code/ChainSafe/git-repos/chainbridge-solidity/scripts/cli/index.js:35:22)
    at process._tickCallback (internal/process/next_tick.js:68:7)
  reason: 'missing argument in Contract constructor',
  code: 'MISSING_ARGUMENT',
  count: 2,
  expectedCount: 3 }
```

I kinda think the output from your changes is better because you have less JavaScript noise in the error message?

Reply:
Yeah I agree it looks a bit nicer. Lets leave it as it?

Fwiw you can validate the exit code by running `echo $?` directly after.